File: sdks/typescript/src/response_format_values.ts
export type JsonSchemaSpec = Readonly<{
  name: string;
  schema: unknown;
  strict: boolean;
}>;

export type ResponseFormat =
  | Readonly<{ type: "text" }>
  | Readonly<{ type: "json_object" }>
  | Readonly<{ type: "json_schema"; jsonSchema: JsonSchemaSpec }>;

/** Response format values; JSON schema validation remains provider-owned. */
export const ResponseFormat = Object.freeze({
  text(): ResponseFormat { return Object.freeze({ type: "text" }); },
  jsonObject(): ResponseFormat { return Object.freeze({ type: "json_object" }); },
  jsonSchema(name: string, schema: unknown): ResponseFormat {
    return Object.freeze({ type: "json_schema", jsonSchema: Object.freeze({ name, schema, strict: true }) });
  },
  isJson(format: ResponseFormat): boolean { return format.type === "json_object" || format.type === "json_schema"; },
});

File: sdks/python/src/echo_agent_sdk/response_format_values.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True, slots=True)
class JsonSchemaSpec:
    name: str
    schema: Any
    strict: bool = True


@dataclass(frozen=True, slots=True)
class ResponseFormat:
    """Response format values; schema validation remains provider-owned."""

    format_type: str
    json_schema_spec: JsonSchemaSpec | None = None

    @property
    def type(self) -> str:
        return self.format_type

    @classmethod
    def text(cls) -> ResponseFormat:
        return cls("text")

    @classmethod
    def json_object(cls) -> ResponseFormat:
        return cls("json_object")

    @classmethod
    def json_schema(cls, name: str, schema: Any) -> ResponseFormat:
        return cls("json_schema", JsonSchemaSpec(name, schema, True))

    def is_json(self) -> bool:
        return self.format_type in {"json_object", "json_schema"}
